lang/cs/Org.Apache.REEF.Client/API/JobRequestBuilder.cs

```diff
@@ -18,25 +18,40 @@
 using System;
 using System.Collections.Generic;
 using Org.Apache.REEF.Common.Client.Parameters;
+using Org.Apache.REEF.Client.Common;
+using Org.Apache.REEF.Client.Common.Parameters;
 using Org.Apache.REEF.Tang.Annotations;
 using Org.Apache.REEF.Tang.Interface;
 using Org.Apache.REEF.Utilities.Logging;
+using System.IO;
+using System.Linq;
 
 namespace Org.Apache.REEF.Client.API
 {
     public sealed class JobRequestBuilder
     {
         private readonly AppParametersBuilder _appParametersBuilder = AppParametersBuilder.NewBuilder();
         private readonly JobParametersBuilder _jobParametersBuilder = JobParametersBuilder.NewBuilder();
+        private const string DLLFileNameExtension = ".dll";
+        private const string EXEFileNameExtension = ".exe";
 
         private JobRequestBuilder()
         {
         }
 
         [Inject]
-        internal JobRequestBuilder([Parameter(typeof(DriverConfigurationProviders))] ISet<IConfigurationProvider> configurationProviders)
+        internal JobRequestBuilder(
+            [Parameter(typeof(DriverConfigurationProviders))] ISet<IConfigurationProvider> configurationProviders,
+            [Parameter(typeof(GlobalAssemblies))] IList<string> globalAssemblies)
         {
             AddDriverConfigurationProviders(configurationProviders);
+
+            // By default, add the current directory
+            if (globalAssemblies.Count == 0)
+            {
+                globalAssemblies.Add(Directory.GetCurrentDirectory());
```

Comment:
This is problematic. If a user doesn't want the local directory added, how can they achieve that?

How about we change semantics  in `.Submit` to:
```cs
if(globalAssemblies.Count == 0 && NoFilesAddedToJobRequest)
{
  // Add the current working directory & the directory of the current executuable
}
```

Reply:
I moved this to the `Build()` statement of the `JobRequest`.

However, it must be noted that this *could* break compatibility with existing applications if they add assemblies *and* expect the local directory to be added by default.